components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -5263,7 +5263,7 @@ private ServiceProviderProperty buildAllowedRoleAudienceProperty(ServiceProvider
         allowedRoleAudienceProperty.setName(ALLOWED_ROLE_AUDIENCE_PROPERTY_NAME);
         AssociatedRolesConfig associatedRolesConfig = serviceProvider.getAssociatedRolesConfig();
         if (associatedRolesConfig == null) {
-            allowedRoleAudienceProperty.setValue(RoleConstants.ORGANIZATION);
+            allowedRoleAudienceProperty.setValue(RoleConstants.APPLICATION);
```

Comment:
Shall we check this change with the legacy runtime as well

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/5663/commits